### PR TITLE
Implement get_filler_item_name for various games

### PR DESCRIPTION
**Pull request by Alchav**
Implements the get_filler_item_name function for various games. Junk items are determined with the following methods:

ALTTP: From the "extras" lists for the specified item pool difficulty
ArchipIDLE: Any random item
FF1: 50% Heal, 25% Pure, 25% Soft
Hollow Knight: Geo Rock-Default
Minecraft: pulls from the junk weights used to normally fill out the item pool with junk
OoT: results of get_junk_item()
Rogue Legacy: random item from misc_items_table
Super Metroid: Nothing
Super Mario 64: 1Up Mushroom
Secret of Evermore: random item from the Ingredients item name group
Slay the Spire: 60% Card Draw, 40% Relic

I'd be happy to make any adjustments / remove sections if any maintainers, etc do not like these methods

### CODE REVIEW

File: worlds/alttp/__init__.py
Comment (Ijwu) [inline worlds/alttp/__init__.py:407]: Could perhaps be shortened to;

```suggestion
            extras_list = difficulties[self.world.difficulty[self.player]].extras[0:4]
            item = self.world.random.choice(extras_list)
```

**Review by Ijwu — COMMENTED** [review]
Seems alright, although I'd be curious as to whether all game implementers are okay with the choices made here. Have you had any discussion with each maintainer about what you went with on these implementations?

**Comment by Alchav** [comment]
> Seems alright, although I'd be curious as to whether all game implementers are okay with the choices made here. Have you had any discussion with each maintainer about what you went with on these implementations?

Nope :)

**Comment by LegendaryLinux** [comment]
ArchipIDLE is valid for any random item.

I definitely want explicit approval from every developer this impacts.

I'm also curious why some games are not included in this list, like Subnautica. Since this function is currently only used by ALttP, I'm unsure what purpose it has in other games.

**Comment by Alchav** [comment]
> ArchipIDLE is valid for any random item.
> 
> I definitely want explicit approval from every developer this impacts.
> 
> I'm also curious why some games are not included in this list, like Subnautica. Since this function is currently only used by ALttP, I'm unsure what purpose it has in other games.

get_filler_item_name is called to fill in when you have item links and do not specify a replacement item, but it has only been implemented for alttp and Timespinner so far.

I did not add it to every game because there are some games I'm not familiar with enough or because I wasn't sure if there were any items that would make sense or that wouldn't cause problems having extras

**Comment by ThePhar** [comment]
It's my understanding that if there are empty locations left, this function is called correct? If that's the case these will need to be removed from the __init__.py file for RL otherwise it'll never be called.

line: 68
```
        total_required_locations = 64 + (self.world.chests_per_zone[self.player] * 4) + (self.world.fairy_chests_per_zone[self.player] * 4)
```

line: 145
```
        for _ in range(len(itempool), total_required_locations):
            item = self.world.random.choice(list(misc_items_table.keys()))
            itempool += [self.create_item(item)]
```

**Comment by Berserker66** [comment]
LttP looks alright, could be made a bit more performant by precomputing the list once or using itertools.chain, I think.
Factorio is missing but that's fair, it has no good filler items.

File: worlds/oot/__init__.py
Comment (espeon65536) [inline worlds/oot/__init__.py:988]: This will not respect the modified junk fills imposed by the ice trap options. You should pass the result of `get_junk_pool(self)` as the `pool` argument to `get_junk_item`, so that you are drawing from the pools with increased ice traps.

**Review by espeon65536 — CHANGES_REQUESTED** [review]

File: worlds/alttp/__init__.py
Comment (black-sliver) [reply worlds/alttp/__init__.py:407]: this does not seem like it should produce the same result:
1. the `+` in the original code merges the lists
2. `[0:4]` does not include `[4]`

**Comment by black-sliver** [comment]
While I raised my concerns regarding the need of this function for item_link already, SoE placing ingredients gets a :+1: from me (until we have energy core hunt, maybe)

File: worlds/alttp/__init__.py
Comment (Ijwu) [reply worlds/alttp/__init__.py:407]: Good catch, my off the cuff python is rusty. 

Updated suggestion:

from itertools import chain
---
extras_list = chain(difficulties[self.world.difficulty[self.player]].extras[0:5]

Either way, I believe that snippet can be improved and I think we might as well if we can.

**Comment by dewiniaid** [comment]
I approve for HK, though the following are also candidates if you want to broaden the pool some:

- All of the `Geo_Rock-*` family other than `Geo_Rock-Outskirts420`
- `Soul_Refill` or `Soul_Totem_A` through `Soul_Totem_G`
- `Geo_Chest-Junk_Pit_1` through `Geo_Chest-Junk_Pit_3`  and `Geo_Chest-Junk_Pit_5` 
- `One_Geo`
- `Lumafly_Escape`  (this is the "Nothing?" item)

**Comment by lordlou** [comment]
For SM, maybe an option to choose between Nothing and Missile would be desirable?
Also, I was pinged for SM64 instead of @N00byKing

**Comment by Alchav** [comment]
> For SM, maybe an option to choose between Nothing and Missile would be desirable? Also, I was pinged for SM64 instead of @N00byKing

I am thinking I will change it to random ammo based on settings. My rationale was that extra ammo can make a big difference in difficulty. However, if someone wants Nothing they can just put Nothing, and there would likely be people interested in filling their game out with extra random ammo

File: worlds/sm64ex/__init__.py
Comment (N00byKing) [inline worlds/sm64ex/__init__.py:87]: Seems fine to me

File: worlds/oot/__init__.py
Comment (Alchav) [reply worlds/oot/__init__.py]: Done

**Comment by Alchav** [comment]
Ran into mod errors when receiving some filler items in HK. Not sure if it's because of receiving the same "specific" geo rocks multiple times or what and things were inconsistent, so I removed HK's function for now.

Changed FF1 to equal chance of each of the 6 consumable items @jtoyoda

**Comment by Alchav** [comment]
@lordlou I have it looking at minors qty setting and using it as a % chance to pick a real item vs nothing. Then looks at your weights for missiles, supers, and power bombs and picking one at random based on the weights

File: worlds/alttp/__init__.py
Comment (Alchav) [reply worlds/alttp/__init__.py]: I don't understand this chain thing and what you wrote alone did not work

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**
